.github/workflows/dockerfiles.yaml

```diff
@@ -18,94 +18,99 @@ on:
     branches:
       - main
 
+
 jobs:
   update:
-    name: Update Dockerfiles
+    name: Test Updated Dockerfiles
     runs-on: ubuntu-latest
     outputs:
       dockerfile_matrix: ${{ steps.dockerfile_list.outputs.dockerfilelist_matrix }}
       empty_matrix: ${{ steps.dockerfile_list.outputs.dockerfilelist_matrix_empty }}
 
     steps:
 
+      # START PULL REQUEST / PUSH checkout and file derivation =======================
       # Using git in the container to diff means we explicitly need to checkout a branch
       # if we use the default of this action we get a detached HEAD state.
       - uses: actions/checkout@v2
-        if: github.event_name == 'pull_request'
+        if: github.event_name != 'schedule'
         with:
            fetch-depth: 0
-           ref: ${{ github.event.pull_request.head.ref }}
 
+      - name: Get changed files
+        id: changed-files
+        if: github.event_name != 'schedule'
+        uses: tj-actions/changed-files@32d6fb28f082f273420574e9de0a30be993bba58       
+        with:
+          separator: " "
+          files_ignore: .github*
+
+      - name: View Changed Files
+        if: github.event_name != 'schedule'
+        run: |
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            echo "$file was changed"
+          done            
+
+      - name: List all Dockerfile and save to matrix
+        uses: vsoch/uptodate@main
+        if: github.event_name != 'schedule'
+        id: dockerfile_list_pr
+        with: 
+          root: ${{ steps.changed-files.outputs.all_changed_files }}
+          parser: dockerfilelist
+          flags: "--no-empty-build-args"
+      # END PULL REQUEST / PUSH checkout and file derivation =======================
+
+      # START SCHEDULED checkout and file derivation =======================
       - uses: actions/checkout@v2
-        if: github.event_name != 'pull_request'
+        if: github.event_name == 'schedule'
         with:
            fetch-depth: 0
+           ref: ${{ github.event.pull_request.head.ref }}
```

Comment:
Does this mean fetch the pull request head on scheduled runs?

Reply:
oh that should probably be removed. It was there for when uptodate was trying to manage the git history.

Reply:
Removed.